library/java/net/openid/appauth/IdToken.java

```diff
@@ -182,7 +182,11 @@ static IdToken from(String token) throws JSONException, IdTokenException {
         }
         final Long expiration = claims.getLong(KEY_EXPIRATION);
         final Long issuedAt = claims.getLong(KEY_ISSUED_AT);
-        final String nonce = JsonUtil.getStringIfDefined(claims, KEY_NONCE);
+        String tempNonce = JsonUtil.getStringIfDefined(claims, KEY_NONCE);
+        if (tempNonce == null) {
+            tempNonce = JsonUtil.getStringIfDefined(headers, KEY_NONCE);
```

Comment:
Please add a comment that we observe the nonce coming from Azure in the header and not the claim.

Reply:
added the comment